src/coreclr/jit/codegenlinear.cpp

```diff
@@ -1657,7 +1657,7 @@ void CodeGen::genConsumeRegs(GenTree* tree)
 #ifdef FEATURE_SIMD
             // (In)Equality operation that produces bool result, when compared
             // against Vector zero, marks its Vector Zero operand as contained.
-            assert(tree->OperIsLeaf() || tree->IsSIMDZero());
+            assert(tree->OperIsLeaf() || tree->IsSIMDZero() || tree->IsVectorZero());
```

Comment:
Would this additional assert be necessary after https://github.com/dotnet/runtime/issues/63490 ?

Reply:
If we replace the initializing tree with `Vector128.Zero` then it would become a leaf node.

Reply:
I guess I'm trying to do that here: https://github.com/dotnet/runtime/pull/63821

How should we proceed? Should be combine the two PRs? I recommend not since there are issues to work out in https://github.com/dotnet/runtime/pull/63821

Reply:
I think we shouldn't combine them.